File: leetcode/src/java/RemoveDuplicatesFromSortedArray.java
public class RemoveDuplicatesFromSortedArray {
    public static void main(String[] args) {
        RemoveDuplicatesFromSortedArray instance = new RemoveDuplicatesFromSortedArray();
        System.out.println(instance.removeDuplicates(new int[]{1,1,2}));
        System.out.println(instance.removeDuplicates(new int[]{0,0,1,1,1,2,2,3,3,4}));
    }
    public int removeDuplicates(int[] nums) {
        int result = 1;
        if (nums.length > 1) {
            for (int i = 1, n = nums.length; i < n; i++) {
                if (nums[i] != nums[i - 1]) {
                    result++;
                    nums[result - 1] = nums[i];
                }
            }
        }
        return result;
    }
}

File: leetcode/src/python/remove_duplicates_from_sorted_array.py
from typing import List

class RemoveDuplicatesFromSortedArray:
    def removeDuplicates(self, nums: List[int]) -> int:
        result = len(nums)
        for i in range(result - 1, 0, -1):
            if nums[i] == nums[i - 1]:
                nums.pop(i)
                result -= 1
        return result

if __name__ == '__main__':
    instance = RemoveDuplicatesFromSortedArray()
    nums = [0,0,1,1,1,2,2,3,3,4]
    print(instance.removeDuplicates(nums))